lumicks/pylake/recording.py

```diff
@@ -0,0 +1,154 @@
+import numpy as np
+import os
+import re
+
+from .channel import Slice, Continuous, TimeSeries
+
+class TiffFrame:
+    """Thin wrapper around a TIFF frame stack. For camera videos timestamps are stored in the DateTime tag in
+    the format start:end.
+
+    Parameters
+    ----------
+    page : tifffile.tifffile.TiffPage
+        Tiff page recorded from a camera in Bluelake.
+    """
+    def __init__(self, page):
+        self._src = page
+
+    @property
+    def data(self):
+        return self._src.asarray()
+
+    @property
+    def start(self):
+        timestamp_string = re.search('^(\d+):\d+$', self._src.tags['DateTime'].value)
```

Comment:
I'd prefer keeping the regex. Some TIFFs may have a date field that is somewhat similar and would match the split (I had a split before). This would generate odd start and stops which would only cause problems further down the line breaking the code in non-intuitive ways. I actually ran into this problem myself opening a different TIFF.

I want to make sure that we actually find our spec specifically and otherwise end up with an empty start and stop.

Reply:
👍